cpp/bridges_module/bridges_test.cpp

```diff
@@ -0,0 +1,213 @@
+#include <queue>
+
+#include <gtest/gtest.h>
+#include <mg_generate.hpp>
+#include <mg_graph.hpp>
+#include <mg_test_utils.hpp>
+
+#include "algorithm/bridges.hpp"
+
+namespace {
+
+std::uint64_t CountComponents(const mg_graph::Graph<> &graph) {
+  std::uint64_t n_components = 0;
+  std::vector<bool> visited(graph.Nodes().size());
+
+  for (auto &node : graph.Nodes()) {
+    if (visited[node.id]) continue;
+    ++n_components;
+
+    std::queue<std::uint64_t> queue;
+    visited[node.id] = true;
+    queue.push(node.id);
+    while (!queue.empty()) {
+      auto curr_id = queue.front();
+      queue.pop();
+      for (const auto &neigh : graph.Neighbours(curr_id)) {
+        auto next_id = neigh.node_id;
+        if (visited[next_id]) continue;
+        visited[next_id] = true;
+        queue.push(next_id);
+      }
+    }
+  }
+  return n_components;
+}
+
+std::vector<std::pair<std::uint64_t, std::uint64_t>> GetBridgesBruteforce(mg_graph::Graph<> *graph) {
+  using IntPair = std::pair<std::uint64_t, std::uint64_t>;
+
+  std::uint64_t comp_cnt = CountComponents(*graph);
+  std::vector<IntPair> bridges;
+  auto edges = graph->ExistingEdges();
+  for (const auto &edge : edges) {
+    auto from = edge.from, to = edge.to;
```

Comment:
```suggestion
  for (const auto &[id, from, to] : edges) {
```

Reply:
For primitive types, like here, it's better to prefer copying over creating a reference.

Reply:
What is the example of copying here?

Reply:
`for (const auto [id, from, to] : edges)`

Reply:
The idea is to remove the reference, since it makes no sense in referencing primitive types, like this:
```suggestion
  for (const auto [id, from, to] : edges) {
```